File: Belajar/Looping/array.php
<?php
$nim = readline("Masukkan NIM: ");

$digit = [];
for ($i = 0; $i < strlen($nim); $i++) {
    $digit[] = intval($nim[$i]);
}

$total = 0;
foreach ($digit as $d) {
    $total += $d;
}

$max = $digit[0];
foreach ($digit as $d) {
    if ($d > $max) $max = $d;
}

$min = $digit[0];
foreach ($digit as $d) {
    if ($d < $min) $min = $d;
}

$rata = $total / count($digit);

$rev = [];
for ($i = count($digit) - 1; $i >= 0; $i--) {
    $rev[] = $digit[$i];
}

echo "digit     [" . implode(", ", $digit) . "]\n";
echo "total     $total\n";
echo "max       $max\n";
echo "min       $min\n";
echo "rata-rata $rata\n";
echo "reverse   [" . implode(", ", $rev) . "]\n";

File: Belajar/Looping/array.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    string nim;
    cout << "Masukkan NIM: ";
    cin >> nim;

    vector<int> digit;
    for (char c : nim) {
        digit.push_back(c - '0');
    }

    int total = 0;
    for (int d : digit) total += d;

    int maks = digit[0];
    for (int d : digit)
        if (d > maks) maks = d;

    int minim = digit[0];
    for (int d : digit)
        if (d < minim) minim = d;

    double rata = (double) total / digit.size();

    vector<int> rev;
    for (int i = digit.size() - 1; i >= 0; i--)
        rev.push_back(digit[i]);

    cout << "digit     ";
    for (int d : digit) cout << d << " ";
    cout << "\ntotal     " << total;
    cout << "\nmax       " << maks;
    cout << "\nmin       " << minim;
    cout << "\nrata-rata " << rata;
    cout << "\nreverse   ";
    for (int r : rev) cout << r << " ";
}
